modules/mono/editor/GodotTools/GodotTools/Internals/GodotSharpDirs.cs

```diff
@@ -79,13 +83,81 @@ public static void DetermineProjectLocation()
             // The csproj should be in the same folder as project.godot.
             string csprojParentDir = "res://";
 
-            _projectSlnPath = Path.Combine(ProjectSettings.GlobalizePath(slnParentDir),
-                string.Concat(_projectAssemblyName, ".sln"));
+            _projectSlnPath = FindSolutionFileWithAssemblyName(slnParentDir, _projectAssemblyName);
 
             _projectCsProjPath = Path.Combine(ProjectSettings.GlobalizePath(csprojParentDir),
                 string.Concat(_projectAssemblyName, ".csproj"));
         }
 
+        private static string FindSolutionFileWithAssemblyName(string directory, string assemblyName)
+        {
+            string globalizedPath = ProjectSettings.GlobalizePath(directory);
+            string[] solutionFiles = Directory.GetFiles(globalizedPath, "*.sln");
+
+            if (solutionFiles.Length == 0)
+            {
+                return Path.Combine(globalizedPath, $"{assemblyName}.sln");
+            }
+
+            List<string> matchingSolutions = new List<string>();
+
+            foreach (string solutionPath in solutionFiles)
+            {
+                try
+                {
+                    var solution = SolutionFile.Parse(solutionPath);
+
+                    foreach (var project in solution.ProjectsInOrder)
+                    {
+                        if (project.ProjectType == SolutionProjectType.SolutionFolder)
+                            continue;
+
+                        if (File.Exists(project.AbsolutePath))
+                        {
+                            var projectRoot = ProjectRootElement.Open(project.AbsolutePath);
+                            var assemblyNameProperty = projectRoot.Properties
+                                .FirstOrDefault(p => p.Name.Equals("AssemblyName", StringComparison.OrdinalIgnoreCase));
+
+                            if (assemblyNameProperty != null &&
+                                assemblyNameProperty.Value.Equals(assemblyName, StringComparison.OrdinalIgnoreCase))
+                            {
+                                matchingSolutions.Add(solutionPath);
+                                break;
+                            }
+
+                            if (assemblyNameProperty == null)
+                            {
+                                string projectNameWithoutExtension = Path.GetFileNameWithoutExtension(project.AbsolutePath);
+                                if (projectNameWithoutExtension.Equals(assemblyName, StringComparison.OrdinalIgnoreCase))
+                                {
+                                    matchingSolutions.Add(solutionPath);
+                                    break;
+                                }
+                            }
```

Comment:
Instead of doing all of this, we can probably just compare the project path (`project.AbsolutePath`) with `_projectCsProjPath`.